### ISSUE TITLE
Add virtual host without recreating container

### ISSUE THREAD

**Issue by sandrodz** [jwilder/nginx-proxy#1051]
Is it possible to add virtual host domains without recreating nginx-proxy container?

**Comment by perarg** [comment]
Each one of these virtual hosts normally should have different root folder (VIRTUAL_ROOT), right ? So maybe a better idea is to have a new container for each one ?

**Comment by sandrodz** [comment]
Its not 1 to 1 thing. Sometimes you've containers with multiple domains, sometimes you've 1 to 1.
Depends on what your trying to achieve right? Having 1 nginx per WordPress project for example is a waste of resources, as 1 can be configured to handle 10 lets say.

**Comment by perarg** [comment]
yes @sandrodz you are right. In case you need multiple virtual hosts i think there is no other way than rebuild your docker container, but this should be confirmed by someone more experienced

**Comment by dantman** [comment]
I have the same issue. I'd like to dump a bunch of basic websites into a single LAMP stack container and deploy newer (e.g. Node.js based applications) to other containers. But it would be pretty annoying to have to recreate the LAMP container and take down all the sites for the whole container launch time instead of just doing a graceful restart.

**Comment by i3c-cloud** [comment]
Occurred to be hard to achieve without modyfying templates or soft. Finally I'm just adding conf files into /etc/nginx/conf.d/ of proxy container + nginx -s reload ...

**Comment by JamBalaya56562** [comment]
Hello, just a third party going through some old issues. For what it's worth, this appears to already be the core, built-in behavior of nginx-proxy rather than something that requires recreating the container. As stated in the current README (v1.11.2): "docker-gen generates reverse proxy configs for nginx and reloads nginx when containers are started and stopped." Internally (see app/Procfile), docker-gen runs with `-watch ... -event-filter event=connect -event-filter event=disconnect -notify "nginx -s reload"`, so when you start a new container with a `VIRTUAL_HOST`, nginx-proxy detects the Docker event, regenerates the config, and issues a graceful `nginx -s reload` — the nginx-proxy container itself is never recreated and existing sites stay up. So adding a new virtual host should be as simple as starting the new container, with no downtime for the others. If you (or the OP) can confirm this matches your current experience on a recent release, maintainers might consider this a candidate for closing, though that's of course their call.

**Comment by buchdag** [comment]
You can't change the environment variables of a container without re-creating it.

If you need to add a new domain to an existing container's `VIRTUAL_HOST` without downtime, you can spawn a new container identical to the first one with the updated `VIRTUAL_HOST` then stop the first one once the second one is up.

The new container will pickup the traffic for the new domain added to the `VIRTUAL_HOST`, and the traffic for the existing domains will be load balanced between the two containers while both are up, then only routed to the second one once the first one has been stoped.
